### PR TITLE
228 comments section

### PR DESCRIPTION
**Pull request by yishain11**

### CODE REVIEW

**Review by avirankatz — CHANGES_REQUESTED** [review]
- [ ] Move comments query to section context
- [x] Remove all console logs
- [ ] We should use a single comments container (before the array rendering)
- [ ] Shorten calcTimeAgoFromDate

File: frontend/src/utils/functions.ts [hunk 1]
```diff
@@ -60,3 +60,58 @@ export function getRemainingSupporters(version: Version) {
...
+  const minDiff = currentMinutes - givenMinutes;
+  if (minDiff > 0) {
+    return `${minDiff} minutes ago`;
+  }
+
+  const secDiff = currentSecs - givenSecs;
+  if (secDiff > 0) {
+    return `${secDiff} seconds ago`;
+  }
+}
```
[the remark this answers is not in the document]
Comment (yishain11) [reply frontend/src/utils/functions.ts:117]: cool implementation!

File: frontend/src/pages/Section.tsx [hunk 2]
```diff
@@ -25,25 +26,43 @@ import {
...
+    variables: {
+      section_version_id: displayedVersion?.id || -1,
+    },
+  };
 
   const navigate = useNavigate();
 
+  useEffect(() => {
+    if (commentVars.variables.section_version_id !== -1) {
+      addComment!(commentVars);
```
[the remark this answers is not in the document]
Comment (yishain11) [reply frontend/src/pages/Section.tsx:62]: this fix results in ts error: 
Object is possibly 'undefined'.ts(2532)
Parsing error: Identifier expected.eslint

same with only ?

File: frontend/src/pages/Section.tsx [hunk 3]
```diff
@@ -25,25 +26,43 @@ import {
...
+    }
+  }, [displayedVersion]);
```
[the remark this answers is not in the document]
Comment (yishain11) [reply frontend/src/pages/Section.tsx:64]: causes infinite loop

File: frontend/src/pages/Section.tsx [hunk 4]
```diff
@@ -190,6 +209,45 @@ const Section: FC = () => {
           </CardContent>
         </Card>
       )}
+      {displayedVersion && (
+        <Card variant="elevation" elevation={0}>
+          <CardContent>
+            <Container maxWidth="sm">
+              {comments &&
+                comments.core_comments.map((comment: Comment) => {
```
[the remark this answers is not in the document]
Comment (yishain11) [reply frontend/src/pages/Section.tsx:217]: done

File: frontend/src/pages/Section.tsx [hunk 5]
```diff
@@ -190,6 +209,45 @@ const Section: FC = () => {
...
+                  return (
+                    <Stack
+                      key={comment.id}
+                      direction="row"
+                      spacing={4}
+                      sx={{ marginBottom: '2rem' }}
```
[the remark this answers is not in the document]
Comment (yishain11) [reply frontend/src/pages/Section.tsx:223]: done

**Review by avirankatz — APPROVED** [review]

**Pull request merged**